packages/core/saved-objects/core-saved-objects-server/src/model_version.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { SavedObjectMigrationContext } from './migration';
+import type { SavedObjectUnsanitizedDoc } from './serialization';
+import type { SavedObjectsMappingProperties } from './mapping_definition';
+
+/**
+ * Root structure defining a model version for a given saved object type.
+ *
+ * model version are mostly described by the operations necessary to evolve the model from the prior to the next
+ * version, and going through the temporary version.
+ */
+export interface SavedObjectsModelVersion<
+  PreviousAttrs = unknown,
+  CompatAttrs = unknown,
+  NextAttrs = unknown
+> {
+  expand: SavedObjectsModelVersionExpand<PreviousAttrs, CompatAttrs>;
+  contract: SavedObjectsModelVersionContract<CompatAttrs, NextAttrs>;
+}
+
+export interface SavedObjectsModelVersionExpand<PreviousAttrs = unknown, CompatAttrs = unknown> {
+  /**
+   * An (optional) list of new fields that will be added to the mapping during the expand phase.
+   */
+  addedMappings?: SavedObjectsMappingProperties;
```

Comment:
This is the subsets of the mappings that will have to be added for this version during the expand phase. 

See the `tag` type examples for more details.